### PR TITLE
Workoff workers added when there are up to three pending long jobs

### PR DESCRIPTION
**Pull request by yakloinsteak**
This does not cap workoff workers spun up entirely because of long jobs to 3. It caps it to three per arbiter run.

Let me know if you want me to change that. The logic will be a little more complicated.

### CODE REVIEW

**Comment by yakloinsteak** [comment]
An update to the deploy secrets goes with this. I'll push that when this is merged into pre-release (it configures only QA at this point unless we just want to run with all of staging which is a minor additional change)

File: app/jobs/workoff_arbiter.rb [hunk 1]
```diff
@@ -61,12 +61,40 @@ def add_worker!
...
 
+  def _long_work_pending?
+    _long_jobs.any?
+  end
+
+  def _num_workers_to_add
+    if _long_work_pending?
+      [_long_jobs.count, 3].min
+    else
+      1
```
Comment (eanders) [inline app/jobs/workoff_arbiter.rb]: Is this going to mean to mean it adds a worker every time, or am I missing something? I read this as if there are no long jobs pending, add a worker.

File: app/jobs/workoff_arbiter.rb [hunk 2]
```diff
@@ -61,12 +61,40 @@ def add_worker!
...
+  def _long_jobs
+    @_long_jobs ||=
+      begin
+        # This prevents this code from silently failing to work correctly if we
+        # were to change the queue name in the future
+        queues = Delayed::Worker.queue_attributes.keys
+        long_queue = 'long_running'
+        raise "Incorrect queue: #{long_queue}" unless long_queue.in?(queues)
+
+        Delayed::Job.where(queue: long_queue, locked_by: nil, last_error: nil, failed_at: nil)
```
Comment (eanders) [inline app/jobs/workoff_arbiter.rb:93]: I'm tempted to invert this to `where.not(queue: 'short_running')` Or maybe drop the queue check entirely, just spool them up if we have jobs in the queue that aren't being worked on.

File: app/jobs/workoff_arbiter.rb (hunk 1 above)
Comment (yakloinsteak) [reply app/jobs/workoff_arbiter.rb]: the add worker method only runs when needs_worker? is true, so it's more of a default of adding one _when we are going to add_

File: app/jobs/workoff_arbiter.rb (hunk 2 above)
Comment (yakloinsteak) [inline app/jobs/workoff_arbiter.rb]: mailers would qualify then, right? Let's just talk about it I think.

Comment (eanders) [reply app/jobs/workoff_arbiter.rb]: Let's ignore the `mailers` queue when determining if we should spin up a worker.  The short running should handle mailers.

Arbiter class will determine length of queue as any unstarted jobs in any queue except `mailers`. `Delayed::Job.where.not(queue: 'mailers').where(locked_by: nil, last_error: nil, failed_at: nil)`

**Comment by yakloinsteak** [comment]
@eanders How does this look now, assuming mailers go to the two short workers?

**Review by eanders — APPROVED** [review]
Looks great to me!

**Comment by yakloinsteak** [comment]
@eanders I think this is ready, and I've merged the secrets yml and pushed in the ansible repo already (the changes are for all of staging). You know what's up with the failing tests?

**Pull request merged**